ballerina/tests/11-custom-object-type-test.bal

```diff
@@ -169,126 +138,299 @@ isolated function insertObjectTypeWithEmptyArray() returns sql:Error? {
+   groups:["execute","object"],
    dependsOn: [insertObjectTypeWithStringArray]
 }
 isolated function insertObjectTypeWithNestedType() returns sql:Error? {
-    Client oracledbClient = check new(HOST, USER, PASSWORD, DATABASE, PORT);
-
     string string_attr = "Hello world";
     int int_attr = 34;
     float float_attr = 34.23;
     decimal decimal_attr = 34.23;
-
     anydata[] attributes = [ string_attr,[string_attr, int_attr, float_attr, decimal_attr]];
     ObjectTypeValue objectType = new({typename: "nested_type", attributes: attributes});
-
     sql:ParameterizedQuery insertQuery = `INSERT INTO TestNestedObjectTypeTable(COL_NESTED_OBJECT)
         VALUES(${objectType})`;
-    sql:ExecutionResult result = check oracledbClient->execute(insertQuery);
-
+    sql:ExecutionResult result = check executeQuery(insertQuery);
     test:assertExactEquals(result.affectedRowCount, 1, "Affected row count is different.");
     var insertId = result.lastInsertId;
     test:assertTrue(insertId is string, "Last Insert id should be string");
+}
+
+type ObjectRecord record {
+    string string_attr;
+    int int_attr;
+    float float_attr;
+    decimal decimal_attr;
+};
+
+type ObjectRecordType record {
+    int pk;
+    ObjectRecord col_object;
+};
+
+@test:Config {
+    groups: ["query", "object"],
+    enable: false,
+    dependsOn: [insertObjectTypeWithNestedType]
+}
+isolated function selectObjectType() returns error? {
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<record{}, error> streamResult = oracledbClient->query(
+        "SELECT pk, col_object FROM TestObjectTypeTable WHERE pk = 1", ObjectRecordType);
+    stream<ObjectRecordType, sql:Error> streamData = <stream<ObjectRecordType, sql:Error>>streamResult;
+    record {|ObjectRecordType value;|}? data = check streamData.next();
+    check streamData.close();
+    ObjectRecordType? value = data?.value;
+    if (value is ()) {
+        test:assertFail("Returned data is nil");
+    } else {
+        test:assertEquals(value.length(), 2);
+        test:assertEquals(value["pk"], 1);
+
+        ObjectRecord objRecord = value["col_object"];
+        decimal delta = 0.01;
+
+        test:assertEquals(objRecord["string_attr"], "Hello world");
+        test:assertEquals(objRecord["int_attr"], 34);
+        test:assertEquals(objRecord["float_attr"], 34.23);
+        test:assertTrue(objRecord["decimal_attr"] - <decimal>34.23 < delta);
+        test:assertTrue(objRecord["decimal_attr"] - <decimal>34.23 > -delta);
+    }
+    check oracledbClient.close();
+}
+
+@test:Config {
+    groups: ["query", "object"],
+    enable: false,
+    dependsOn: [selectObjectType]
+}
+isolated function selectObjectTypeNull() returns error? {
+    Client oracledbClient = check new (HOST, USER, PASSWORD, DATABASE, PORT);
+    stream<record{}, error> streamResult = oracledbClient->query(
+        "SELECT pk, col_object FROM TestObjectTypeTable WHERE pk = 15", ObjectRecordType);
```

Comment:
Let's always try to go with the type infering from the assigned variable type. As it is our recommendation.

Reply:
corrected in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/7/commits/903f0cd0b25dea933b9106c0947ee5a969e13afd